aea/configurations/loader.py

```diff
@@ -202,6 +204,198 @@ def _validate(self, json_data: Dict) -> None:
+        :return: the processed component configuration.
+        """
+        component_id, result = self._split_component_id_and_config(
+            i, component_configuration_json
+        )
+        self._validate_component_configuration(component_id, result)
+        self._check_only_configurable_fields(component_id, result)
+        return component_id, result
+
+    @staticmethod
+    def _split_component_id_and_config(
+        i: int, component_configuration_json: Dict
+    ) -> Tuple[ComponentId, Dict]:
+        """
+        Split component id and configuration.
+
+        :param i: the position of the component configuration in the agent config file..
+        :param component_configuration_json: the JSON object to process.
+        :return: the component id and the configuration object.
+        :raises ValueError: if the component id cannot be extracted.
+        """
+        result = deepcopy(component_configuration_json)
+        # author, name, version, type are mandatory fields
+        missing_fields = {"author", "name", "version", "type"}.difference(
+            component_configuration_json.keys()
+        )
+        if len(missing_fields) > 0:
+            raise ValueError(
+                f"There are missing fields in component id {i + 1}: {missing_fields}."
+            )
+        component_name = result.pop("name")
+        component_author = result.pop("author")
+        component_version = result.pop("version")
+        component_type = ComponentType(result.pop("type"))
+        component_public_id = PublicId(
+            component_author, component_name, component_version
+        )
+        component_id = ComponentId(component_type, component_public_id)
+        return component_id, result
+
+    @staticmethod
+    def _validate_component_configuration(
+        component_id: ComponentId, configuration: Dict
+    ) -> None:
+        """
+        Validate the component configuration of an agent configuration file.
+
+        This check is to detect inconsistencies in the specified fields.
+
+        :param component_id: the component id.
+        :param configuration: the configuration dictionary.
+        :return: None
+        :raises ValueError: if the configuration is not valid.
+        """
+        # we need to populate the required fields to validate the configurations.
+        temporary_config = deepcopy(configuration)
+        # common to every package
+        temporary_config["name"] = component_id.name
+        temporary_config["author"] = component_id.author
+        temporary_config["version"] = component_id.version
+        temporary_config["license"] = "some_license"
+        temporary_config["aea_version"] = "0.1.0"
+        if component_id.component_type == ComponentType.PROTOCOL:
+            pass  # no other required field
+        elif component_id.component_type == ComponentType.CONNECTION:
+            temporary_config["class_name"] = "SomeClassName"
```

Comment:
what's going on here? also, declare constant, no magic var

Reply:
see above, please.

Reply:
probably create a set of dummy configs for every component that should pass validation and update those configs with variables from agent config and check will.loader? also requires tests for that configs validation

Reply:
@solarw thanks that sounds like a good idea